EIPS/eip-5164.md

```diff
@@ -0,0 +1,209 @@
+
+MUST increment a `nonce` so that each batch of calls can be uniquely identified.
+
+MAY require payment. Some bridges may require payment in the native currency, so the function is payable.
+
+```solidity
+struct Call {
+  address target;
+  bytes data;
+}
+
+interface CrossChainRelayer {
+    function relayCalls(CrossChainExecutor executor, Call[] calldata calls, uint256 gasLimit) external payable;
+}
+```
+
+```yaml
+- name: relayCalls
+  type: function
+  stateMutability: payable
+  inputs:
+    - name: executor
+      type: CrossChainExecutor
+    - name: calls
+      type: Call[]
+    - name: gasLimit
+      type: uint256
+```
+
+#### Events
+
+**RelayedCalls**
+
+MUST be emitted by the `CrossChainRelayer` when `relayCalls` is called.
+
+```solidity
+interface CrossChainRelayer {
+  event RelayedCalls(
+      uint256 indexed nonce,
+      address indexed sender,
+      CrossChainExecutor indexed executor,
+      Call[] calls,
+      uint256 gasLimit
+  );
+}
+```
+
+```yaml
+- name: RelayedCalls
+  type: event
+  inputs:
+    - name: nonce
+      indexed: true
+      type: uint256
+    - name: sender
+      indexed: true
+      type: address
+    - name: executor
+      indexed: true
+      type: address
+    - name: calls
+      type: Call[]
+    - name: gasLimit
+      type: uint256
+```
+
+#### Error handling
+
+**GasLimitTooHigh**
+
+MAY revert if the `gasLimit` passed to `relayCalls` is higher than the maximum gas limit accepted by the bridge being used.
+
+```solidity
+interface CrossChainRelayer {
+  error GasLimitTooHigh(
+    uint256 gasLimit,
+    uint256 maxGasLimit
+  );
+}
+```
+
+### CrossChainExecutor
+
+The `CrossChainExecutor` executes relayed calls. Developers must implement a `CrossChainExecutor` in order to execute messages on the receiving chain.
+
+#### Authentication
+
+SHOULD authenticate that the call has been performed by the bridge transport layer.
+
+SHOULD follow [EIP-2771](./eip-2771.md) and append the address of the Transaction Signer (20 bytes of data) on the origin chain to the end of the call data being executed.
+
+#### Error handling
+
+**CallFailure**
+
+SHOULD revert if a call fails.
+
+```solidity
+interface CrossChainExecutor {
+  error CallFailure(
+    Call call,
+    bytes errorData
+  );
+}
+```
+
+#### Events
+
+**ExecutedCalls**
+
+MUST be emitted once calls have been executed.
+
+```solidity
+interface CrossChainExecutor {
+  event ExecutedCalls(
+    CrossChainRelayer indexed relayer,
+    uint256 indexed nonce,
+    address indexed caller
+  );
+}
+```
+
+```yaml
+- name: ExecutedCalls
+  type: event
+  inputs:
+    - name: relayer
+      indexed: true
+      type: CrossChainRelayer
+    - name: nonce
+      indexed: true
+      type: uint256
+    - name: caller
+      indexed: true
+      type: address
+```
+
+### ExecutorAware
+
+The `ExecutorAware` follows [EIP-2771](./eip-2771.md) and allows applications to verify where the call originated from.
```

Comment:
```suggestion
The `ExecutorAware` interface follows [EIP-2771](./eip-2771.md) and allows applications to verify where the call originated from.
```

Reply:
Fixed in this commit: https://github.com/ethereum/EIPs/pull/5164/commits/a478089e0a4e04f9d0fcaff2b5c95169bc5af23c#diff-41efe0617f861cc7330b025ecc86d55fe777d3c503208d83dca04258de4f5f41R180